hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -805,51 +805,73 @@ public boolean isBucketEmpty(String volume, String bucket)
     return deletedKeys;
   }
 
+  /**
+   * @param userName volume owner, null for listing all volumes.
+   * @throws IOException
+   */
   @Override
   public List<OmVolumeArgs> listVolumes(String userName,
       String prefix, String startKey, int maxKeys) throws IOException {
     List<OmVolumeArgs> result = Lists.newArrayList();
-    UserVolumeInfo volumes;
+    List<UserVolumeInfo> volumes = Lists.newArrayList();
+
     if (StringUtil.isBlank(userName)) {
-      throw new OMException("User name is required to list Volumes.",
-          ResultCodes.USER_NOT_FOUND);
+      // null userName represents listing all volumes in cluster.
+      for (String user : getAllUsers()) {
```

Comment:
Thanks for catching this issue and fixing it.

With HA now we have Table Cache. So, when the request is received we add it to the cache and then return the result. So, when we want to get all users in the system, we want to merge the data in User table cache and DB.

For a particular user list of Volumes is stored in UserVolumeInfo. So, there is no issue in that part.

For more info, you can refer to a recent fix that has been done in this part for listBuckets API (https://issues.apache.org/jira/browse/HDDS-1984). In a similar way, you need to change getAllUsers() to get all users from cache and DB. As now we want to support listAllVolumes in the system.

Reply:
Thanks @adoroszlai for the previous comments about ```getVolumeTable``` and 
thanks @bharatviswa504 for this comments for the ```cache``` about ```getAllUsers```.

I fixed this part just for ```getAllUsers``` from cache & DB, and it works.
(The UT needs no change cause the ```TestOMRequestUtils#addVolumeToOM``` involving the cache & DB)

I think we can fix/implement ```getVolumeByAdmin()``` with cache & DB like this.